src/native/libs/System.Native/pal_dynamicload.c

```diff
@@ -57,13 +57,18 @@ void* SystemNative_GetDefaultSearchOrderPseudoHandle(void)
     return (void*)RTLD_DEFAULT;
 }
 #else
-static void* g_defaultSearchOrderPseudoHandle = NULL;
+volatile static void* g_defaultSearchOrderPseudoHandle = NULL;
```

Comment:
```suggestion
static volatile void* g_defaultSearchOrderPseudoHandle = NULL;
```

Reply:
The typical convention for `volatile` is to keep it closer to the type rather than the linkage use of `static`.